src/metabase/util/stats.clj

```diff
@@ -0,0 +1,332 @@
+            [clj-http.client :as client]
+            (metabase [config :as config]
+                      [db :as db])
+            [metabase.public-settings :as settings]
+            (metabase.models [field :as field]
+                             [table :as table]
+                             [setting :as setting])
+            [metabase.util :as u]))
+
+(def ^:private ^:const ^String metabase-usage-url "https://kqatai1z3c.execute-api.us-east-1.amazonaws.com/prod/ServerStatsCollector")
+
+(def ^:private ^Integer anonymous-id
+  "Generate an anonymous id. Don't worry too much about hash collisions or localhost cases, etc.
+   The goal is to be able to get a rough sense for how many different hosts are throwing a specific error/event."
+  (hash (str (java.net.InetAddress/getLocalHost))))
+
+
+(defn- anon-tracking-enabled?
+  "To avoid a circular reference"
+  []
+  (require 'metabase.public-settings)
+  (resolve 'metabase.public-settings/anon-tracking-enabled))
+
+
+
+(defn- bin-micro-number
+  "Return really small bin number. Assumes positive inputs"
+  [x]
+  (cond
+    (= 0 x) "0"
+    (= 1 x) "1"
+    (= 2 x) "2"
+    (> x 2) "3+")
+  )
+
+
+(defn- bin-small-number
+  "Return small bin number. Assumes positive inputs"
+  [x]
+  (cond
+    (= 0 x) "0"
+    (<= 1 x 5) "1-5"
+    (<= 6 x 10) "6-10"
+    (<= 11 x 25) "11-25"
+    (> x 25) "25+")
+  )
+
+(defn- bin-medium-number
+  "Return medium bin number. Assumes positive inputs"
+  [x]
+  (cond
+    (= 0 x) "0"
+    (<= 1 x 5) "1-5"
+    (<= 6 x 10) "6-10"
+    (<= 11 x 25) "11-25"
+    (<= 26 x 50) "26-50"
+    (<= 51 x 100) "51-100"
+    (<= 101 x 250) "101-250"
+    (> x 250) "250+")
+  )
+
+
+(defn- bin-large-number
+  "Return large bin number. Assumes positive inputs"
+  [x]
+  (cond
+    (= 0 x) "0"
+    (< x 1) "< 1"
+    (<= 1 x 10) "1-10"
+    (<= 11 x 50) "11-50"
+    (<= 51 x 250) "51-250"
+    (<= 251 x 1000) "251-1000"
+    (<= 1001 x 10000) "1001-10000"
+    (> x 10000) "10000+")
+  )
+
+(defn- get-value-frequencies
+  "go through a bunch of maps and count the frequency a given key's values"
+  [many-maps k]
+  (frequencies (map k many-maps))
+  )
+
+(defn- histogram
+  "Bin some frequencies using a passed in binning function"
+  [binning-fn many-maps k]
+  (frequencies (map binning-fn (vals (get-value-frequencies many-maps k))))
+  )
+
+(def micro-histogram
+  "return a histogram for micro numbers"
+  (partial histogram bin-micro-number))
+
+(def small-histogram
+  "return a histogram for small numbers"
+  (partial histogram bin-small-number))
+
+(def medium-histogram
+  "return a histogram for medium numbers"
+  (partial histogram bin-medium-number))
+
+(def large-histogram
+  "return a histogram for large numbers"
+  (partial histogram bin-large-number))
+
+(defn- instance-start-date?
+  "Pull up the first user account and use that date"
+  []
+  ((first (db/select 'User {:limit 1} {:order-by [[:date_joined :desc]]})) :date_joined)
+  )
+
+(defn- where-am-i-running?
```

Comment:
I would name function this something like `environment-type` because a function ending in `?` implies that it's a predicate 

https://github.com/bbatsov/clojure-style-guide#pred-with-question-mark